jina/serve/runtimes/gateway/__init__.py

```diff
@@ -79,6 +78,8 @@ async def async_setup(self):
             uses_metas={},
             runtime_args={  # these are not parsed to the yaml config file but are pass directly during init
                 'name': self.args.name,
+                'port': self.args.port,
```

Comment:
why is it moved to the runtime args now ?

Reply:
port and protocol are now parameters that the runtime is aware of (used for health checks, k8s service, docker compose expose, ...)
The user no more has control over this parameter and he can't define how this parameter looks like, it is provided by the runtime for him

Reply:
so the user cannot change the port for example ?

Reply:
He passes a port as input according to our runtime defined format:
```python
Flow().config_gateway(port=12345)
```
or
```python
Flow().config_gateway(port=[12345, 12344])
```
and will read it from runtime_args when he implements the gateway.
He cannot come up with his own format or be able to customize it, like
```python
Flow().config_gateway(port={'http': 12345, 'grpc': 12344)
```

Reply:
okay I see makes sense